.github/workflows/build-deploy-dev.yml

```diff
@@ -5,42 +5,133 @@ on:
     types: [opened, reopened, synchronize, closed]
     branches:
       - main
+  workflow_dispatch:
+    inputs:
+      bumb:
+        description: "The type of semantic version increment to make. By default it is 'prerelease'."
+        required: false
+        default: "prerelease"
+        type: choice
+        options:
+          - major
+          - premajor
+          - minor
+          - preminor
+          - patch
+          - prepatch
+          - prerelease
+      preId:
+        description: "Identifier to be used to prefix version increments. By default it is 'prerelease' for pull request event and workflow dispatch will follow the last release tag prefix, if any."
```

Comment:
Can this one be derived from the previous one?

Reply:
ya, we can. I thought if at any point, we want to create any `preX` bump tag without a `preX` prefix/prefid, we have the option to do that, right?
But I liked the idea of creating the prefix from the bump. it will make the workflow dispatch 1 less field (which might confuse the user). i will give it a try